### PR TITLE
Fix next task not scheduled after switch completes

### PR DESCRIPTION
**Pull request by jxu-nflx**
Pull Request type
----
- [x] Bugfix
- [ ] Feature
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes (Please run `./gradlew generateLock saveLock` to refresh dependencies)
- [ ] WHOSUSING.md
- [ ] Other (please describe):

**NOTE**: Please remember to run `./gradlew spotlessApply` to fix any format violations.

Changes in this PR
----
Revert changes done to SwitchTaskMapper introduced by [this PR](https://github.com/Netflix/conductor/commit/2d514690eb7a52ec6140677a1bf083f6f940443d#diff-c3a45903d808c1c4aa854c8fd387310c24e9d8fcee276392093ca40168e18f88), and add test case to verify the fix.

_Describe the new behavior from this PR, and why it's needed_
Issue #
https://github.com/Netflix/conductor/issues/3089#issuecomment-1224034537

Alternatives considered
----

_Describe alternative implementation you have considered_

### CODE REVIEW

**Review by apanicker-nflx — APPROVED** [review]

**Comment by BrandonDotLin** [comment]
@jxu-nflx Would you please kindly advise If this PR fixed only switch task or applicable for all kinds of tasks? i.e. fixed the issue #3089 completely?

**Comment by v1r3n** [comment]
> @jxu-nflx Would you please kindly advise If this PR fixed only switch task or applicable for all kinds of tasks? i.e. fixed the issue #3089 completely?

Yes this will fix #3089

**Comment by BrandonDotLin** [comment]
> > @jxu-nflx Would you please kindly advise If this PR fixed only switch task or applicable for all kinds of tasks? i.e. fixed the issue #3089 completely?
> 
> Yes this will fix #3089

So great. Looking forward to the release including this PR.

File: core/src/main/java/com/netflix/conductor/core/execution/mapper/SwitchTaskMapper.java [hunk 1]
```diff
@@ -96,8 +96,7 @@ public List<TaskModel> getMappedTasks(TaskMapperContext taskMapperContext) {
         switchTask.getInputData().put("case", evalResult);
         switchTask.getOutputData().put("evaluationResult", Collections.singletonList(evalResult));
         switchTask.setStartTime(System.currentTimeMillis());
-        switchTask.setStatus(TaskModel.Status.COMPLETED);
-        switchTask.setExecuted(true);
+        switchTask.setStatus(TaskModel.Status.IN_PROGRESS);
```
Comment (manan164) [inline core/src/main/java/com/netflix/conductor/core/execution/mapper/SwitchTaskMapper.java:99]: The switch task is completed here so the status should be `COMPLETED` .Only executed flag has to be removed.

Comment (jxu-nflx) [reply core/src/main/java/com/netflix/conductor/core/execution/mapper/SwitchTaskMapper.java:99]: It will be marked as completed in the executor

**Comment by jxu-nflx** [comment]
> @jxu-nflx Would you please kindly advise If this PR fixed only switch task or applicable for all kinds of tasks? i.e. fixed the issue #3089 completely?

This is the fix just for the switch task

**Pull request merged**